tests/Unit.hs

```diff
@@ -178,106 +178,119 @@ rangeTests = describe "range tests" $ do
-        ("1.2.3+asdf - 2.4.3+asdf", "1.2.3"),
-        ("1.0.0", "1.0.0"),
-        (">=*", "0.2.4"),
-        ("", "1.0.0"),
-        ("*", "1.2.3"),
-        ("*", "v1.2.3"),
-        (">=1.0.0", "1.0.0"),
-        (">=1.0.0", "1.0.1"),
-        (">=1.0.0", "1.1.0"),
-        (">1.0.0", "1.0.1"),
-        (">1.0.0", "1.1.0"),
-        ("<=2.0.0", "2.0.0"),
-        ("<=2.0.0", "1.9999.9999"),
-        ("<=2.0.0", "0.2.9"),
-        ("<2.0.0", "1.9999.9999"),
-        ("<2.0.0", "0.2.9"),
-        (">= 1.0.0", "1.0.0"),
-        (">=  1.0.0", "1.0.1"),
-        (">=   1.0.0", "1.1.0"),
-        ("> 1.0.0", "1.0.1"),
-        (">  1.0.0", "1.1.0"),
-        ("<=   2.0.0", "2.0.0"),
-        ("<= 2.0.0", "1.9999.9999"),
-        ("<=  2.0.0", "0.2.9"),
-        ("<    2.0.0", "1.9999.9999"),
-        ("<\t2.0.0", "0.2.9"),
-        (">=0.1.97", "v0.1.97"),
-        (">=0.1.97", "0.1.97"),
-        ("0.1.20 || 1.2.4", "1.2.4"),
-        (">=0.2.3 || <0.0.1", "0.0.0"),
-        (">=0.2.3 || <0.0.1", "0.2.3"),
-        (">=0.2.3 || <0.0.1", "0.2.4"),
-        ("||", "1.3.4"),
-        ("2.x.x", "2.1.3"),
-        ("1.2.x", "1.2.3"),
-        ("1.2.x || 2.x", "2.1.3"),
-        ("1.2.x || 2.x", "1.2.3"),
-        ("x", "1.2.3"),
-        ("2.*.*", "2.1.3"),
-        ("1.2.*", "1.2.3"),
-        ("1.2.* || 2.*", "2.1.3"),
-        ("1.2.* || 2.*", "1.2.3"),
-        ("*", "1.2.3"),
-        ("2", "2.1.2"),
-        ("2.3", "2.3.1"),
-        ("~2.4", "2.4.0"),
-        ("~2.4", "2.4.5"),
-        ("~>3.2.1", "3.2.2"),
-        ("~1", "1.2.3"),
-        ("~>1", "1.2.3"),
-        ("~> 1", "1.2.3"),
-        ("~1.0", "1.0.2"),
-        ("~ 1.0", "1.0.2"),
-        ("~ 1.0.3", "1.0.12"),
-        (">=1", "1.0.0"),
-        (">= 1", "1.0.0"),
-        ("<1.2", "1.1.1"),
-        ("< 1.2", "1.1.1"),
-        ("~v0.5.4-pre", "0.5.5"),
-        ("~v0.5.4-pre", "0.5.4"),
-        ("=0.7.x", "0.7.2"),
-        ("<=0.7.x", "0.7.2"),
-        (">=0.7.x", "0.7.2"),
-        ("<=0.7.x", "0.6.2"),
-        ("~1.2.1 >=1.2.3", "1.2.3"),
-        ("~1.2.1 =1.2.3", "1.2.3"),
-        ("~1.2.1 1.2.3", "1.2.3"),
-        ("~1.2.1 >=1.2.3 1.2.3", "1.2.3"),
-        ("~1.2.1 1.2.3 >=1.2.3", "1.2.3"),
-        ("~1.2.1 1.2.3", "1.2.3"),
-        (">=1.2.1 1.2.3", "1.2.3"),
-        ("1.2.3 >=1.2.1", "1.2.3"),
-        (">=1.2.3 >=1.2.1", "1.2.3"),
-        (">=1.2.1 >=1.2.3", "1.2.3"),
-        (">=1.2", "1.2.8"),
-        ("^1.2.3", "1.8.1"),
-        ("^0.1.2", "0.1.2"),
-        ("^0.1", "0.1.2"),
-        ("^1.2", "1.4.2"),
-        ("^1.2 ^1", "1.4.2"),
-        ("^1.2.3-alpha", "1.2.3-pre"),
-        ("^1.2.0-alpha", "1.2.0-pre"),
-        ("^0.0.1-alpha", "0.0.1-beta")
+  let testCases :: [(Bool, Text, Text)] = [
+
+        (True, "^1.2.3-alpha.1", "1.2.3-alpha.7"),
+        (True, "^0.0.1-alpha.1", "0.0.1-alpha.t"),
+        (True, "^0.0.1-alpha.1", "0.0.1-alpha.tdff.dddddddddd"),
+        (False, "^1.2.3-alpha", "2.0.0-alpha"),
```

Comment:
The `False` examples involving prerelease tags are the more interesting ones IMO. In the cases that they're testing distinct corner cases, it would be nice to have a comment explaining the particular reason why they should fail

Reply:
Sure! I'll try to keep the comments brief.